### DISCUSSION TITLE
Zustand and Custom hooks.

### DISCUSSION THREAD

**Discussion by Gblur** [neuefische/questions#564] [Web]
Hey everyone,

I got a problem when using Zustand store with custom hooks.

reproduce: 
Click and hold mousebutton on the node handler and drop the newly created line anywhere in the canvas.
Currently App does not work.

logic:
I destructured the useNodeCreator hook, to get access to the onConnectEnd Function.
This returns two Objects {newNode, newEdge}
Calling the set() function onConnectEnd to spread over previous nodes and add a newNode by adding the Object onConnectEnd.newNode

result: 
Cannot read properties of null (reading 'useRef')

Link PR: 
https://github.com/Gblur/capstone-project/pull/6

The Error comes from nodeCreator.js line 6

![Screenshot 2023-04-05 at 11 20 38](https://user-images.githubusercontent.com/25586795/230038667-49b4d419-73e1-4cc0-87c4-000c942d4f4e.png)

Error Message: 

![Screenshot 2023-04-05 at 11 45 26](https://user-images.githubusercontent.com/25586795/230044862-43018922-c236-4557-b4e5-e18e375bffa9.png)

**Comment by JessicaLoers** [comment]
**Hey @Gblur** ✌️

This is an interesting problem. To be able to help you, provide the community with the link to the affected branch / file / line ✨

**Comment by Gblur** [comment]
Updated my question.

**Comment by doemser** [comment]
Not so easy to identify your challenge.

You import your custom hook into your state store. This is super awkward and I suspect it could break your app, or rather, cause a consequential error.

Zustand stores are not React components and do not have access to the React component lifecycle. When you import a React hook into a Zustand store, it may not behave as expected or could cause unexpected side effects.

For now, try structuring your app to use the hook from a react component. 

I think this will eliminate the error you are currently facing, due to restructuring you code.

**Reply by Gblur** [reply]
thx for the answer.
Thats what I thought, so I tried to initialize the useref hook in the component where I imported the store.
Dont know how Zustand can pass Params.
Can I use something like a placeholder in the store, and not directly pass the value for the setter of onConnectStart, onConnectend. 
Something like onConnectStart: param
isnt there another way then the useRef hook to pass the currentID?

Update: 
Maybe I found a method for solving this problem. 
Just wrap the store into the custom hook and so everything is in scope and accessible

Update:
Seems not to work after all. I willclose this thread as unresolved and try a different approach
